streams/src/main/java/org/apache/kafka/streams/processor/internals/SourceNode.java

```diff
@@ -29,14 +30,20 @@
     private ProcessorContext context;
     private Deserializer<K> keyDeserializer;
     private Deserializer<V> valDeserializer;
+    private final TimestampExtractor timestampExtractor;
 
-    public SourceNode(String name, List<String> topics, Deserializer<K> keyDeserializer, Deserializer<V> valDeserializer) {
+    public SourceNode(String name, List<String> topics, TimestampExtractor timestampExtractor, Deserializer<K> keyDeserializer, Deserializer<V> valDeserializer) {
```

Comment:
@jeyhunkarimov Hmm, in `TopologyBuilder#build` when we are adding the source node we will execute the pattern matching from the current topic metadata so the map should already be filled with the actual topic right?

I think the problem, as I added in `SourceNodeFactory#getTopics`'s comment, is that under debugging / unit testing environment, there is no topic metadata available to pass the topic list to `List<String> getTopics(Collection<String> subscribedTopics)`. What we need to do to fix the unit tests then is to call `SubscriptionUpdates#updateTopics()` before calling `builder.build()` for regex involved tests.

```
                    final List<String> topics = (sourceNodeFactory.pattern != null) ?
                            sourceNodeFactory.getTopics(subscriptionUpdates.getUpdates()) :
                            sourceNodeFactory.topics;

                    for (String topic : topics) {
                        if (internalTopicNames.contains(topic)) {
                            // prefix the internal topic name with the application id
                            topicSourceMap.put(decorateTopic(topic), (SourceNode) node);
                        } else {
                            topicSourceMap.put(topic, (SourceNode) node);
                        }
                    }
```